builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -697,18 +704,15 @@ func (p *provisioner) runCommand(o terraform.UIOutput, comm communicator.Communi
 	}
 
 	cmd.Wait()
```

Comment:
Yes, I was going to do the same thing, but then decided to try and keep the API changes minimal, while getting rid of the races and adding an error value. It turns out that I probably would have had the same amount of changes to handle had I added an error value to Wait, with a specific `ExitError` error type for the return code. 🤷‍♂️ 

I think we can make that a separate patch, since it's more cosmetic than anything at this point.